packages/xo-server-web-hooks/src/index.js

```diff
@@ -0,0 +1,134 @@
+import createLogger from '@xen-orchestra/log'
+import { groupBy, mapValues } from 'lodash'
+
+const log = createLogger('xo:web-hooks')
+
+function makeRequest(url, type, data) {
+  return this._xo.httpRequest(url, {
+    body: JSON.stringify({ ...data, type }),
+    headers: { 'Content-Type': 'application/json' },
+    method: 'POST',
+    onRequest: req => {
+      req.setTimeout(1e4)
+      req.on('timeout', req.abort)
+    },
+  })
+}
+
+function handleHook(type, data) {
+  let hooks
+  if ((hooks = this._conf[data.method]?.[type]) !== undefined) {
+    return Promise.all(
+      hooks.map(({ url }) =>
+        this._makeRequest(url, type, data).catch(log.error)
+      )
+    )
+  }
+}
+
+class XoServerHooks {
+  constructor({ xo }) {
+    this._xo = xo
+
+    // Defined in configure().
+    this._conf = null
+
+    this._handlePreHook = handleHook.bind(this, 'pre')
+    this._handlePostHook = handleHook.bind(this, 'post')
+    this._makeRequest = makeRequest.bind(this)
+  }
+
+  configure(configuration) {
+    // this._conf = {
+    //   'vm.start': {
+    //     pre: [
+    //       {
+    //         method: 'vm.start',
+    //         type: 'pre',
+    //         url: 'https://my-domain.net/xo-hooks?action=vm.start'
+    //       },
+    //       ...
+    //     ],
+    //     post: [
+    //       ...
+    //     ]
+    //   },
+    //   ...
+    // }
+    this._conf = mapValues(groupBy(configuration, 'method'), _ =>
```

Comment:
IMHO, it will be better to do it manually to decrease the number of loops which will impact the performance.